### PR TITLE
Improve TzIf format detection

### LINKED ISSUE #214: Time zone info format detection could be more accurate
Right now the check is:
https://github.com/gabriel-vasile/mimetype/blob/222d00f101284dcaaf19a485e02665d27fcc9728/internal/magic/binary.go#L21
This is a lot of room for false positives with this check.

Solution: 
https://tools.ietf.org/id/draft-murchison-tzdist-tzif-00.html#rfc.section.3

### PR DESCRIPTION
**Pull request by n-vr**
Added the four-octet ASCII sequence to identify TzIf files: `0x54 0x5A 0x69 0x66`
[Source](https://tools.ietf.org/id/draft-murchison-tzdist-tzif-00.html#rfc.section.3.p.4)

> This pull request  fixes #214

### CODE REVIEW

**Review by gabriel-vasile — COMMENTED** [review]
Hi, thank you for taking your time to contribute.

The problem I tried to describe in #214 is that the detection is too loose. Any simple text file that happens to start with `TZif` is detected as `application/tzif` (see the commit I added on this PR, it shows the problem.)

https://tools.ietf.org/id/draft-murchison-tzdist-tzif-00.html#rfc.section.3 describes in detail the format of the tzif file. For example, fifth byte, `version`, must be 0x00, 0x32 or 0x33, `charcnt` bytes must not be 0, etc.
These are the checks that we can do for tzif files to make the detection more accurate.

**Comment by n-vr** [comment]
Sorry for overwriting twice, my signature was not working properly because of a new system install.

File: internal/magic/binary.go [hunk 1]
```diff
@@ -155,3 +153,39 @@ func Marc(raw []byte, limit uint32) bool {
...
+	}
+
+	if !bytes.HasPrefix(raw, []byte("TZif")) {
+		return false
+	}
+
+	// Field "typecnt" MUST not be zero.
+	if bytes.Equal(raw[36:40], []byte{0x00}) {
+		return false
+	}
```
Comment (gabriel-vasile) [inline internal/magic/binary.go:181]:
```suggestion
	if binary.LittleEndian.Uint32(raw[36:40]) == 0 {
		return false
	}
```
The specification says:
_NOTE: All multi-octet integer values MUST be stored in big-endian format (high-order octet first)._
So you have to interpret those 4 bytes as a big-endian value.

File: mimetype_test.go [hunk 2]
```diff
@@ -597,3 +597,10 @@ func TestExtend(t *testing.T) {
 		})
 	}
 }
+
+func TestTzif(t *testing.T) {
+	mtype, _ := DetectFile("testdata/tzif.txt")
+	if mtype.Is("application/tzif") {
+		t.Fatal("simple text file should not be detected as Tzif file")
+	}
+}
```
Comment (gabriel-vasile) [inline mimetype_test.go:606]: Please remove this test function and the file at testdata/tzif.txt. I added it to better show you the problem and it is not needed anymore.

**Review by gabriel-vasile — CHANGES_REQUESTED** [review]
Hi, code looks better but there are still some issues to fix. I left some suggestions.

You can commit how many times you want, there is not problem.

File: internal/magic/binary.go [hunk 3]
```diff
@@ -155,3 +153,39 @@ func Marc(raw []byte, limit uint32) bool {
...
+
+	// Version has to be NUL (0x00), '2' (0x32) or '3' (0x33).
+	version := raw[4:5]
+	for _, validVersion := range [][]byte{{0x00}, {0x32}, {0x33}} {
+		if bytes.Equal(version, validVersion) {
+			return true
+		}
+	}
+	return false
```
Comment (gabriel-vasile) [inline internal/magic/binary.go:190]:
```suggestion
	return raw[4] == 0x00 || raw[4] == 0x32 || raw[4] == 0x33
```
You don't need to use `bytes.Equal` when comparing a single byte value. The suggested code is easier to read and faster to run.

File: mimetype_test.go (hunk 2 above)
Comment (n-vr) [reply mimetype_test.go]: Removed in commit.

**Review by gabriel-vasile — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**